conformance/utils/kubernetes/apply.go

```diff
@@ -143,13 +143,24 @@ func (a Applier) MustApplyObjectsWithCleanup(t *testing.T, c client.Client, time
 		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.CreateTimeout)
 		defer cancel()
 
-		t.Logf("Creating %s %s", resource.GetName(), resource.GetObjectKind().GroupVersionKind().Kind)
+		t.Logf("Creating/Updating %s %s", resource.GetName(), resource.GetObjectKind().GroupVersionKind().Kind)
 
-		err := c.Create(ctx, resource)
+		// Check if the resource exists
+		existingResource := resource.DeepCopyObject().(client.Object)
+		err := c.Get(ctx, types.NamespacedName{Name: resource.GetName(), Namespace: resource.GetNamespace()}, existingResource)
 		if err != nil {
-			if !apierrors.IsAlreadyExists(err) {
-				require.NoError(t, err, "error creating resource")
+			if !apierrors.IsNotFound(err) {
+				t.Logf("error checking resource: %s %s", resource.GetName(), resource.GetObjectKind().GroupVersionKind().Kind)
+				require.NoError(t, err, "error checking resource")
 			}
+
+			// Resource doesn't exist, create it
+			err = c.Create(ctx, resource)
+			require.NoError(t, err, "error creating resource")
+		} else {
+			// Resource exists, update it
+			err = c.Update(ctx, resource)
```

Comment:
if you update you might want to setResourceVersion
Like MustApplyWithCleanup is doing 

`uObj.SetResourceVersion(fetchedObj.GetResourceVersion())`

Reply:
You can resolve this